### PR TITLE
Add Mapbox Draw feature on GeoInput

### PR DESCRIPTION
**Pull request by tnagorra**
Depends on https://github.com/the-deep/server/pull/429
Depends on https://github.com/toggle-corp/react-store/pull/694
Depends on https://github.com/toggle-corp/react-store/pull/695

### CODE REVIEW

File: src/components/input/GeoInput/GeoModal/index.js [hunk 1]
```diff
@@ -39,7 +42,8 @@ const propTypes = {
 };
 
 const defaultProps = {
-    title: '',
+    // FIXME: use strings
+    title: 'Select geo areas',
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/index.js]: Use strings

File: src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                        >
+                            {/* FIXME: use strings */}
+                            Cancel
+                        </DangerButton>
+                        <PrimaryButton
+                            type="submit"
+                            disabled={pristine}
+                        >
+                            {/* FIXME: use strings */}
+                            Save
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js]: Use strings

File: src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                            // FIXME: use strings
+                            label="Color"
+                        />
+                    </ModalBody>
+                    <ModalFooter>
+                        <DangerButton
+                            onClick={onClose}
+                        >
+                            {/* FIXME: use strings */}
+                            Cancel
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js]: Use strings

File: src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                        <TextInput
+                            faramElementName="title"
+                            // FIXME: use strings
+                            label="Title"
+                            autoFocus
+                        />
+                        <ColorInput
+                            faramElementName="color"
+                            // FIXME: use strings
+                            label="Color"
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js]: Use strings

File: src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                    schema={schema}
+                    value={faramValues}
+                    error={faramErrors}
+                >
+                    <NonFieldErrors faramElement />
+                    <ModalBody>
+                        <TextInput
+                            faramElementName="title"
+                            // FIXME: use strings
+                            label="Title"
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js]: Use strings

File: src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        const {
+            value,
+            onClose,
+        } = this.props;
+
+        return (
+            <Modal>
+                <ModalHeader
+                    // FIXME: use strings
+                    title={`Edit ${value.type}`}
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/PolygonPropertiesModal/index.js]: Use strings

File: src/components/geo/RegionMap/index.js [hunk 7]
```diff
@@ -534,15 +760,46 @@ export default class RegionMap extends React.PureComponent {
...
+                                </DangerButton>
+                            </>
+                        ) : (
+                            <Button
+                                className={styles.button}
+                                onClick={onEditStart}
+                                disabled={myAdminLevelPending}
+                            >
+                                {/* FIXME: use strings */}
+                                Edit points/polygons
```
Comment (tnagorra) [inline src/components/geo/RegionMap/index.js]: Use strings

File: src/components/geo/RegionMap/index.js [hunk 8]
```diff
@@ -534,15 +760,46 @@ export default class RegionMap extends React.PureComponent {
...
+                                    disabled={myAdminLevelPending}
+                                >
+                                    Save
+                                </PrimaryButton>
+                                <DangerButton
+                                    className={styles.button}
+                                    onClick={onEditCancel}
+                                    disabled={myAdminLevelPending}
+                                >
+                                    Cancel
```
Comment (tnagorra) [inline src/components/geo/RegionMap/index.js]: Use strings

File: src/components/geo/RegionMap/index.js [hunk 9]
```diff
@@ -534,15 +760,46 @@ export default class RegionMap extends React.PureComponent {
...
+                            pending={myAdminLevelPending}
+                        />
+                        {editMode ? (
+                            <>
+                                <PrimaryButton
+                                    className={styles.button}
+                                    onClick={this.handleCompleteEditMode}
+                                    disabled={myAdminLevelPending}
+                                >
+                                    Save
```
Comment (tnagorra) [inline src/components/geo/RegionMap/index.js]: Use strings

File: src/components/input/GeoInput/GeoModal/GeoInputList/index.js [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+    } = props;
+
+    const handlePolygonEdit = useCallback(
+        (localId) => {
+            const polygon = polygons.find(p => p.localId === localId);
+            if (!polygon) {
+                console.error('Could not find index for polygon localId', localId);
+                return;
+            }
+            console.warn('Do something for', polygon);
```
Comment (AdityaKhatri) [inline src/components/input/GeoInput/GeoModal/GeoInputList/index.js]: Pls remove sir

File: src/widgetComponents/Matrix1dListOutput/Row/index.js [hunk 11]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 
 import ListView from '#rscv/List/ListView';
 
-import ListItem from '#components/general/ListItem';
+import ListItem from '#rscv/List/ListItem';
```
Comment (AdityaKhatri) [inline src/widgetComponents/Matrix1dListOutput/Row/index.js]: It should be #rscv/ListItem, right?

File: src/widgetComponents/Matrix2dListOutput/Row/index.js [hunk 12]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 import List from '#rscv/List';
-import ListItem from '#components/general/ListItem';
+import ListItem from '#rscv/List/ListItem';
```
Comment (AdityaKhatri) [inline src/widgetComponents/Matrix2dListOutput/Row/index.js]: It should be #rscv/ListItem, right?

File: src/widgetComponents/MultiSelectListOutput/index.js [hunk 13]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 import { FaramOutputElement } from '@togglecorp/faram';
 
 import ListView from '#rscv/List/ListView';
-import ListItem from '#components/general/ListItem';
+import ListItem from '#rscv/List/ListItem';
```
Comment (AdityaKhatri) [inline src/widgetComponents/MultiSelectListOutput/index.js]: It should be #rscv/ListItem, right?

File: src/widgetComponents/OrganigramListOutput/index.js [hunk 14]
```diff
@@ -3,8 +3,7 @@ import PropTypes from 'prop-types';
 import { FaramOutputElement } from '@togglecorp/faram';
 
 import ListView from '#rscv/List/ListView';
-
-import ListItem from '#components/general/ListItem';
+import ListItem from '#rscv/List/ListItem';
```
Comment (AdityaKhatri) [inline src/widgetComponents/OrganigramListOutput/index.js]: It should be #rscv/ListItem, right?

File: src/widgetComponents/Matrix1dListOutput/Row/index.js (hunk 11 above)
Comment (tnagorra) [reply src/widgetComponents/Matrix1dListOutput/Row/index.js]: Oh yes, Correct!

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**